src/main/kotlin/no/nav/familie/ba/mottak/integrasjoner/InfotrygdBarnetrygdClient.kt

```diff
@@ -0,0 +1,105 @@
+import java.net.URI
+import java.time.LocalDate
+
+@Component
+class InfotrygdBarnetrygdClient(@Value("\${FAMILIE_BA_INFOTRYGD_BARNETRYGD_API_URL}/infotrygd/barnetrygd")
+                                private val clientUri: URI,
+                                @Qualifier("clientCredentials") restOperations: RestOperations,
+                                private val environment: Environment)
+    : AbstractRestClient(restOperations, "infotrygd_barnetrygd_replika") {
+
+    fun hentLøpendeUtbetalinger(søkersIdenter: List<String>, barnasIdenter: List<String>): InfotrygdSøkResponse<StønadDto> {
+        return infotrygdResponseFra(
+                request = { postForEntity(uri("stonad"), InfotrygdSøkRequest(søkersIdenter, barnasIdenter)) },
+                onFailure = { ex -> IntegrasjonException("Feil ved søk etter stønad i infotrygd.", ex, uri("stonad")) }
+        )
+    }
+
+    fun hentSaker(søkersIdenter: List<String>, barnasIdenter: List<String>): InfotrygdSøkResponse<SakDto> {
+        return infotrygdResponseFra(
+                request = { postForEntity(uri("saker"), InfotrygdSøkRequest(søkersIdenter, barnasIdenter)) },
+                onFailure = { ex -> IntegrasjonException("Feil ved uthenting av saker fra infotrygd.", ex, uri("saker")) }
+        )
+    }
+
+    private fun <T> infotrygdResponseFra(request: () -> InfotrygdSøkResponse<T>,
+                                         onFailure: (Throwable) -> RuntimeException): InfotrygdSøkResponse<T> {
+        return if (environment.activeProfiles.contains("e2e"))  InfotrygdSøkResponse(emptyList(), emptyList())
+        else runCatching(request).getOrElse { throw onFailure(it) }
+    }
+
+    private fun uri(endepunkt: String) = URI.create("$clientUri/$endepunkt")
+}
+
+data class InfotrygdSøkRequest(val brukere: List<String>,
+                               val barn: List<String>? = null)
+
+data class InfotrygdSøkResponse<T> (
+        val bruker: List<T>,
+        val barn: List<T>,
+)
+
+
+val InfotrygdSøkResponse<StønadDto>.resultat: Sakspart?
+    get() = if (bruker.isNotEmpty()) SØKER else if (barn.isNotEmpty()) ANNEN else null
+
+val InfotrygdSøkResponse<SakDto>.resultat: Sakspart? @JvmName("getSakspart")
+    get() = if (bruker.harSak()) SØKER else if (barn.harSak()) ANNEN else null
+
+private fun List<SakDto>.harSak(): Boolean {
+    val (sakerMedVedtak, sakerUtenVedtak) = this.partition { it.vedtak != null }
+
+    return sakerMedVedtak.let { it.all { it.vedtak!!.opphørsgrunn != Opphørsgrunn.MIGRERT.kode } && it.isNotEmpty() }
+           || sakerUtenVedtak.any { it.status != StatusKode.FB.name }
+}
+
+data class StønadDto(
+        val stønadId: Long,
+        val sakNr: String? = null,
+        val opphørtFom: String? = null,
+        val opphørsgrunn: String? = null,
+)
+
+data class SakDto(
+        val saksnr: String? = null,
+        val kapittelnr: String? = null,
+        val valg: String? = null,
+        val Type: String? = null,
```

Comment:
Type. Aarsakskode, Vedtaksdato, RegAvEnhet, BehenEnhet bryter god navnestandard for kotlin/java. Kanskje fikse og ta i bruk @JsonProperty("Type")

Reply:
👍 fikser det i dto på begge sider